data/scenarios/Challenges/capture.yaml

```diff
@@ -0,0 +1,185 @@
+          if (n > 0) {
+              blockedFwd <- blocked;
+              if blockedFwd {
+                  turn left;
+                  isSurrounded $ n - 1;
+              } {
+                  return false;
+              }
+          } {
+              return true;
+          }
+          end;
+
+      opponent <- robotnamed "opponent";
+      as opponent {
+        isSurrounded 4;
+      };
+  - teaser: Spooked
+    id: spooked
+    optional: true
+    hidden: true
+    goal:
+      - |
+        The pig was spooked by the capture attempt.
+    condition: |
+      def boolToInt = \b.
+          if b {1} {0}
+          end;
+
+      /**
+      Iterate 4 times (once for each direction)
+      */
+      def countBlocked = \n.
+          if (n > 0) {
+              isBlocked <- blocked;
+              // This assignment has to happen before the recursive
+              // "countBlocked" call due to #1032
+              let localBlockCount = boolToInt isBlocked in
+              turn left;
+              lastCount <- countBlocked $ n - 1;
+              return $ lastCount + localBlockCount;
+          } {
+              return 0;
+          }
+          end;
+
+      opponent <- robotnamed "opponent";
+      as opponent {
+          blockCount <- countBlocked 4;
+          return $ blockCount > 0 && blockCount < 4;
+      };
+robots:
+  - name: base
+    dir: [1, 0]
+    devices:
+      - 3D printer
+      - branch predictor
+      - calculator
+      - clock
+      - comparator
+      - dictionary
+      - dozer blade
+      - grabber
+      - hearing aid
+      - lambda
+      - logger
+      - olfactometer
+      - scanner
+      - strange loop
+      - treads
+      - welder
+      - workbench
+    inventory:
+      - [6, dozer blade]
+      - [6, ADT calculator]
+      - [6, branch predictor]
+      - [6, clock]
+      - [6, comparator]
+      - [6, counter]
+      - [6, dictionary]
+      - [6, grabber]
+      - [6, lambda]
+      - [6, lodestone]
+      - [6, logger]
+      - [6, mirror]
+      - [6, net]
+      - [6, rolex]
+      - [6, scanner]
+      - [6, solar panel]
+      - [6, string]
+      - [6, strange loop]
+      - [6, treads]
+      - [6, welder]
+  - name: opponent
+    system: true
+    dir: [0, 1]
+    display:
+      char: P
+      attr: pig
+      invisible: false
+    program: |
+      run "scenarios/Challenges/_capture/opponent.sw"
+solution: |
+   run "scenarios/Challenges/_capture/solution.sw"
+entities:
+  - name: dozer blade
+    display:
+      attr: silver
+      char: '/'
+    description:
+    - Facilitates pushing
+    properties: [known, portable]
+    capabilities: [push]
+  - name: rolex
+    display:
+      char: 'R'
+      attr: silver
+    description:
+      - Enables robots to use the 'watch' command.
+    properties: [known, portable]
+    capabilities: [wakeself]
+  - name: olfactometer 
+    display:
+      char: 'N'
+    description:
+    - Smells the distance to an entity
```

Comment:
Will remove.  Originally I had thought to use this to "smell" the "bacon" placed by the pig, in case the pig runs far away in some unpredictable direction.  But I since have simplified the pig's "escape" logic.